src/00_utils.js

```diff
@@ -0,0 +1,76 @@
+function pushPoly(array, item) {
+  array[array.length] = item;
+}
+
+function slicePoly(array, start, end) {
+  let result = [];
+  let count = 0;
+  const length = array.length;
+  end = end > length ? length : end;
+  end = end === undefined || end === null ? length : end;

```

Comment:
I'd recommend you do not mutate functions arguments and to keep functions clean, you can define a new variable
https://habr.com/ru/articles/437512/